znh5md/io.py

```diff
@@ -19,6 +20,34 @@
 # TODO: use pint to convert the units in the h5md file to ase units
 
 
+def _process_batch(self, batch_slice, index):
+    tmp_images = _getitem(self, batch_slice)
+    return [
+        (idx + batch_slice.start, image)
+        for idx, image in enumerate(tmp_images)
+        if idx + batch_slice.start in index
+    ]
+
+
+def _get_images(self, index, chunk_size):
+    images = []
+    with concurrent.futures.ProcessPoolExecutor() as executor:
+        futures = []
+        for batch in range(0, len(self), chunk_size):
+            batch_slice = slice(batch, batch + chunk_size)
+            futures.append(executor.submit(_process_batch, self, batch_slice, index))
```

Comment:
you can skip batches where the indices are not present

Reply:
The number of batches is typically much smaller than the number of requested samples thus this is not worth it.